test/project_types/script/layers/infrastructure/script_uploader_test.rb

```diff
@@ -0,0 +1,122 @@
+describe Script::Layers::Infrastructure::ScriptUploader do
+  include TestHelpers::Partners
+
+  describe "UploadScript" do
+    let(:ctx) { TestHelpers::FakeContext.new }
+    let(:api_key) { "fake_key" }
+    let(:script_service) { Script::Layers::Infrastructure::ScriptService.new(ctx: ctx, api_key: api_key) }
```

Comment:
For unit testing here, I think we should use a mock object here, and stub out the `module_upload_url_generate`:
`mock.expects(:module_upload_url_generate).returns(script_service_response)`.

Reply:
These tests were just copied out of the `ScriptService` tests to isolate them. I avoided changing anything else for now.